langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiAudioModel.java

```diff
@@ -0,0 +1,411 @@
+import dev.langchain4j.model.output.FinishReason;
+import dev.langchain4j.model.output.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Map;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.model.azure.InternalAzureOpenAiHelper.*;
+import static dev.langchain4j.spi.ServiceHelper.loadFactories;
+
+/**
+ * Represents an OpenAI audio model, hosted on Azure, such as whisper.
+ * <p>
+ * You can find a tutorial on using Azure OpenAI for speech to text at: https://learn.microsoft.com/azure/ai-services/openai/whisper-quickstart
+ * <p>
+ * Mandatory parameters for initialization are: endpoint and apikey (or an alternate authentication method, see below for more information).
+ * Optionally you can set serviceVersion (if not, the latest version is used) and deploymentName (if not, a default name is used).
+ * You can also provide your own OpenAIClient instance, if you need more flexibility.
+ * <p>
+ * By default, the response type will be json with the raw text included.
+ * You can change the response format using the responseFormat() method in the Builder, or the responseFormat parameter in the constructor.
+ * <p>
+ * There are 3 authentication methods:
+ * <p>
+ * 1. Azure OpenAI API Key Authentication: this is the most common method, using an Azure OpenAI API key.
+ * You need to provide the OpenAI API Key as a parameter, using the apiKey() method in the Builder, or the apiKey parameter in the constructor:
+ * For example, you would use `builder.apiKey("{key}")`.
+ * <p>
+ * 2. non-Azure OpenAI API Key Authentication: this method allows to use the OpenAI service, instead of Azure OpenAI.
+ * You can use the nonAzureApiKey() method in the Builder, which will also automatically set the endpoint to "https://api.openai.com/v1".
+ * For example, you would use `builder.nonAzureApiKey("{key}")`.
+ * The constructor requires a KeyCredential instance, which can be created using `new AzureKeyCredential("{key}")`, and doesn't set up the endpoint.
+ * <p>
+ * 3. Azure OpenAI client with Microsoft Entra ID (formerly Azure Active Directory) credentials.
+ * - This requires to add the `com.azure:azure-identity` dependency to your project, which is an optional dependency to this library.
+ * - You need to provide a TokenCredential instance, using the tokenCredential() method in the Builder, or the tokenCredential parameter in the constructor.
+ * As an example, DefaultAzureCredential can be used to authenticate the client: Set the values of the client ID, tenant ID, and
+ * client secret of the AAD application as environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET.
+ * Then, provide the DefaultAzureCredential instance to the builder: `builder.tokenCredential(new DefaultAzureCredentialBuilder().build())`.
+ */
+public class AzureOpenAiAudioModel implements AudioModel {
```

Comment:
```suggestion
public class AzureOpenAiAudioTranscriptionModel implements AudioModel {
```
I would make it a bit more specific, as there is another type of "audio model" that generates speech from text.

Reply:
Yes, we should differentiate Transcription API between others Open AI APIs (Speech API and Translation API)